src/shared/treeview/nodes/errorNode.ts

```diff
@@ -5,13 +5,30 @@
 
 import * as vscode from 'vscode'
 import { AWSTreeNodeBase } from './awsTreeNodeBase'
+import { localize } from '../../../shared/utilities/vsCodeUtils'
+import { getLogger, Logger } from '../../../shared/logger/logger'
+import { getLogTracker, LogTrackerRecord } from '../../../shared/logger/logTracker'
 
 // Used as a child node when an exception occurs while querying AWS resources
 export class ErrorNode extends AWSTreeNodeBase {
     public constructor(public readonly parent: AWSTreeNodeBase, public readonly error: Error, label: string) {
         super(label, vscode.TreeItemCollapsibleState.None)
+        const commandName: string = localize('AWS.command.viewLogs', 'View AWS Toolkit Logs')
+        const tooltip: string = localize('AWS.explorerNode.error.tooltip', 'Click to view error in Toolkit logs')
+        const logger: Logger = getLogger()
+        const logRecord: LogTrackerRecord = getLogTracker().registerLog()
+        logger.error(error, { logID: logRecord.logID })     
```

Comment:
I think I finally understand :) This is a very important piece, no? Essentially :

1. here we are passing a globally unique id to associate with a message. 
2. when the transport writes the message, it publishes an event / calls-back
3. the handler/subscriber can then get the "rasterized"/actual message contents assoc'd with the id

Reply:
Yeah that is a pretty important part. Yup you got the design correct. I started redesigning it before I saw this comment, though I did your first two points on my own. I agree with your third point as well, I think I just got carried away with Promises since I had never used them before. They seem fun.